bpf/lib/nodeport.h

```diff
@@ -1846,6 +1849,12 @@ static __always_inline int nodeport_lb4(struct __ctx_buff *ctx,
 	}
 
 	if (!backend_local) {
+#ifdef ENABLE_WIREGUARD
+		/* The request which is to be forwarded needs to go over the
+		 * Wireguard tunnel.
+		 */
+		ctx->mark |= MARK_MAGIC_ENCRYPT;
+#endif /* ENABLE_WIREGUARD */
```

Comment:
Could we move all of these into small inline helpers that better document the intent? Would ipsec use the same?

Reply:
Do we need to make KPR mandatory for WG?

Reply:
No, KPR is not mandatory. This case is only for encrypting a fwd NodePort BPF request to a remote endpoint.

Reply:
> Would ipsec use the same?

I haven't thoroughly checked the mark usage in our IPSec code.

Reply:
Mark field is also open coded like above in ipsec. Helpers would be nice.